xla/mlir/tools/mlir_replay/public/execution_trace_utils.cc

```diff
@@ -280,6 +280,8 @@ absl::StatusOr<InterpreterValue> LiteralToValue(const xla::Literal& literal) {
         return {{ArrayLiteralToTensor<double>(literal)}};
       case xla::F8E5M2:
         return absl::UnimplementedError("F8E5M2 not implemented");
+      case xla::F8E4M3:
+        return absl::UnimplementedError("F8E4M3 not implemented");
```

Comment:
To avoid having to modify this every time a new FP8 type is added, remove all these FP8 cases and check if `IsF8Type(literal.shape().element_type()` before the switch statement.

Reply:
Opened PR https://github.com/openxla/xla/pull/17170 Code dedup in execution_trace_utils LiteralToValue

Reply:
Just FYI, this was an intentional choice. Missing switch cases are a compiler error, so having a switch without a default case is preferable when possible. No big deal though.

Reply:
Using single-case switch statements can make it easier for the compiler to detect potential errors in the code.
Opened PR https://github.com/openxla/xla/pull/17279 - Use switch case without default in LiteralToValue